### PR TITLE
added NVMeoF modules

### PR DESCRIPTION
**Pull request by sunilkumarn417**
- Added NVMeoF modules for NVMeoF GW deployment.
- Added NVMe-CLI modules for managing sub-system entities.
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-LPEDSI/

- Updated old test cases of IO perf comparison
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DACN98/
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-K9YVJQ/
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-Q21ST8/

### CODE REVIEW

**Comment by rahullepakshi** [comment]
@sunilkumarn417 Do we still need to configure GW node as client as at https://github.com/red-hat-storage/cephci/blob/master/suites/reef/nvmeof/tier-0_nvmeof_sanity.yaml#L58 ?

Any test run logs?  to get the flow

File: suites/reef/nvmeof/tier-0_nvmeof_sanity.yaml [hunk 1]
```diff
@@ -69,6 +69,28 @@ tests:
...
 #  Run IO on NVMe Targets
+  - test:
+      abort-on-fail: true
+      config:
+        verify_cluster_health: true
+        steps:
+          - config:
+              command: shell
+              args:
+                - ceph osd pool create rbd
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply suites/reef/nvmeof/tier-0_nvmeof_sanity.yaml:79]: @rahullepakshi, pool creation is not part of orchestration command

**Comment by sunilkumarn417** [comment]
> @sunilkumarn417 Do we still need to configure GW node as client as at https://github.com/red-hat-storage/cephci/blob/master/suites/reef/nvmeof/tier-0_nvmeof_sanity.yaml#L58 ?
> 
> Any test run logs? to get the flow

Updated i nthe description.

**Review by rahullepakshi — DISMISSED** [review]

**Comment by sunilkumarn417** [comment]
Updated old test cases of IO perf comparison
- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-DACN98/
- http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-K9YVJQ/

**Comment by rahullepakshi** [comment]
Test run logs for 4 scale tests-

1. https://github.com/red-hat-storage/cephci/blob/006de4ac1ec6575f159bb9b729ccea6ff793698a/suites/reef/nvmeof/tier-2_nvmeof_scale_ns.yaml -   http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-EEL4WY/Scale_to_256_namespaces_in_single_subsystem_on_NVMeOF_GW_0.log
2. https://github.com/red-hat-storage/cephci/blob/006de4ac1ec6575f159bb9b729ccea6ff793698a/suites/reef/nvmeof/tier-2_nvmeof_scale_sub_ns.yaml - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-JOPSFQ/
3. https://github.com/red-hat-storage/cephci/blob/006de4ac1ec6575f159bb9b729ccea6ff793698a/suites/reef/nvmeof/tier-2_nvmeof_scale_ns.yaml - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-TCG3MJ
4. https://github.com/red-hat-storage/cephci/blob/006de4ac1ec6575f159bb9b729ccea6ff793698a/suites/reef/nvmeof/tier-2_nvmeof_subsystem_limit.yaml - http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-02F9U2/

**Review by Manohar-Murthy — APPROVED** [review]
lgtm.

**Pull request merged**